daemon/src/daemon.rs

```diff
@@ -830,24 +837,111 @@ impl Daemon {
+                                {
+                                    let bytes = sandbox::read_file(&base_dir_cloned, &path)
+                                        .expect("Failed to read created file");
+                                    let content = String::from_utf8(bytes)
+                                        .expect("Could not read file as UTF-8");
+
+                                    document_handle
+                                        .send_message(DocMessage::CreateFile {
+                                            file_path: path
+                                                .to_str()
+                                                .expect("Failed to convert path to string")
+                                                .into(),
+                                            content,
+                                        })
+                                        .await;
+                                }
+                            }
+                        }
+                        notify::event::EventKind::Modify(notify::event::ModifyKind::Data(_)) => {
+                            for path in event.paths {
+                                if !sandbox::ignored(&base_dir_cloned, &path)
+                                    .expect("Could not determine ignored status of file")
+                                {
+                                    let bytes = sandbox::read_file(&base_dir_cloned, &path)
+                                        .expect("Failed to read created file");
+                                    let content = String::from_utf8(bytes)
+                                        .expect("Could not read file as UTF-8");
+
+                                    document_handle
+                                        .send_message(DocMessage::UpdateFile {
+                                            file_path: path
+                                                .to_str()
+                                                .expect("Failed to convert path to string")
+                                                .into(),
+                                            content,
+                                        })
+                                        .await;
+                                }
+                            }
+                        }
+                        notify::event::EventKind::Modify(notify::event::ModifyKind::Name(
+                            notify::event::RenameMode::Both,
+                        )) => {
+                            let from_path = &event.paths[0];
+                            let to_path = &event.paths[1];
+
+                            // TODO: Avoid repeating the effects here.
+                            // TODO: Is there a cleverer way to do renames in the CRDT?
+
                             document_handle
                                 .send_message(DocMessage::RemoveFile {
```

Comment:
haha this diff confused me. "logically" this is the old removal event above, but that got added instead... just leaving this comment in case we come back here :)